easybuild/framework/extensioneasyblock.py

```diff
@@ -100,18 +100,27 @@ def __init__(self, *args, **kwargs):
         self.ext_dir = None  # dir where extension source was unpacked
 
     def _set_start_dir(self):
-        """Set value for self.start_dir
+        """Set absolute path of self.start_dir similarly to EasyBlock.guess_start_dir
 
-        Uses existing value of self.start_dir if it is already set and exists
-        otherwise self.ext_dir (path to extracted source) if that is set and exists, similar to guess_start_dir
+        Uses existing value of self.start_dir if it is already set, an absolute path and exists
+        otherwise use self.ext_dir (path to extracted source) as base dir if that is set and exists.
         """
-        possible_dirs = (self.start_dir, self.ext_dir)
-        for possible_dir in possible_dirs:
-            if possible_dir and os.path.isdir(possible_dir):
-                self.cfg['start_dir'] = possible_dir
-                self.log.debug("Using start_dir: %s", self.start_dir)
-                return
-        self.log.debug("Unable to determine start_dir as none of these paths is set and exists: %s", possible_dirs)
+        ext_start_dir = ''
+
+        if self.start_dir:
+            ext_start_dir = self.start_dir
+
+        if not os.path.isabs(ext_start_dir) and self.ext_dir:
+            # start dir is either empty or a _relative_ path provided by user through self.start_dir
+            # generate absolute path from ext_dir
+            ext_start_dir = os.path.join(self.ext_dir, ext_start_dir)
+
+        if os.path.isdir(ext_start_dir):
+            self.cfg['start_dir'] = ext_start_dir
+            self.log.debug("Using extension start dir: %s", ext_start_dir)
+        else:
+            # non-existing start dir means wrong input from user
+            self.log.debug("Provided start dir for extension does not exist: %s", ext_start_dir)
```

Comment:
This is as it was before but I'm a bit curious here, why not inform the user that what they thought they were going to use is not going to be used (i.e., make this a warning)?

Reply:
Right now, if I use this and make a typo, I'd have to create a debug log to discover it.

Reply:
There's an example in https://github.com/easybuilders/easybuild-framework/commit/766347b27597413c180e87a6c75f862b29081231 of how that could be caught in the test

Reply:
I'll open a PR for this myself

Reply:
Done in #4201